diffusion/models/models.py

```diff
@@ -580,6 +581,306 @@ def stable_diffusion_xl(
     return model
 
 
+def precomputed_text_latent_diffusion(
+    unet_model_name: str = 'stabilityai/stable-diffusion-xl-base-1.0',
+    vae_model_name: str = 'madebyollin/sdxl-vae-fp16-fix',
+    autoencoder_path: Optional[str] = None,
+    autoencoder_local_path: str = '/tmp/autoencoder_weights.pt',
+    include_text_encoders: bool = False,
+    text_encoder_dtype: str = 'float16',
+    cache_dir: str = '/tmp/hf_files',
+    prediction_type: str = 'epsilon',
+    image_key: str = 'image',
+    t5_latent_key: str = 'T5_LATENTS',
+    t5_mask_key: str = 'T5_ATTENTION_MASK',
+    clip_latent_key: str = 'CLIP_LATENTS',
+    clip_mask_key: str = 'CLIP_ATTENTION_MASK',
+    clip_pooled_key: str = 'CLIP_POOLED',
+    latent_mean: Union[float, Tuple, str] = 0.0,
+    latent_std: Union[float, Tuple, str] = 7.67754318618,
+    text_embed_dim: int = 4096,
+    train_noise_scheduler_params: Optional[Dict[str, Any]] = None,
+    inference_noise_scheduler_params: Optional[Dict[str, Any]] = None,
+    scheduler_shift_resolution: int = 256,
+    train_metrics: Optional[List] = None,
+    val_metrics: Optional[List] = None,
+    quasirandomness: bool = False,
+    train_seed: int = 42,
+    val_seed: int = 1138,
+    fsdp: bool = True,
+    use_xformers: bool = True,
+    lora_rank: Optional[int] = None,
+    lora_alpha: Optional[int] = None,
+):
+    """Latent diffusion model training using precomputed text latents from T5-XXL and CLIP.
+
+    Args:
+        unet_model_name (str): Name of the UNet model to load. Defaults to
+            'stabilityai/stable-diffusion-xl-base-1.0'.
+        vae_model_name (str): Name of the VAE model to load. Defaults to
+            'madebyollin/sdxl-vae-fp16-fix' as the official VAE checkpoint (from
+            'stabilityai/stable-diffusion-xl-base-1.0') is not compatible with fp16.
+        autoencoder_path (optional, str): Path to autoencoder weights if using custom autoencoder. If not specified,
+            will use the vae from `model_name`. Default `None`.
+        autoencoder_local_path (optional, str): Path to autoencoder weights. Default: `/tmp/autoencoder_weights.pt`.
+        include_text_encoders (bool): Whether to include text encoders in the model. Should only do this for running
+            inference. Default: `False`.
+        text_encoder_dtype (str): The dtype to use for the text encoder. One of [`float32`, `float16`, `bfloat16`].
```

Comment:
Perhaps the default value for this should match the default value of `latent_dtype` of `StreamingImageCaptionLatentsDataset`? (diffusion/datasets/image_caption_latents.py on L44 which sets it to bfloat)

Reply:
Fixed!